### PR TITLE
Upgrade sdk dependencies

### LINKED ISSUE #2749: Upgrade SDC and Engine libraries
We now have newer SDC and Engine artifacts from the latest changes on the SDK:

- Engine - `org.smartregister:engine:0.1.0-beta04-preview2-SNAPSHOT`
- SDC - `org.smartregister:data-capture:1.0.0-preview17-SNAPSHOT`

_Originally posted by @ndegwamartin in https://github.com/opensrp/fhircore/issues/2708#issuecomment-1717226524_

### PR DESCRIPTION
**Pull request by ellykits**
**IMPORTANT: Where possible all PRs must be linked to a Github issue**

Fixes #2749

**Engineer Checklist**
- [ ] I have written **Unit tests** for any new feature(s) and edge cases for bug fixes
- [ ] I have added any strings visible on UI components to the `strings.xml` file
- [ ] I have updated the  [CHANGELOG.md](./CHANGELOG.md) file for any notable changes to the codebase
- [ ] I have run `./gradlew spotlessApply` and `./gradlew spotlessCheck` to check my code follows the project's style guide
- [ ] I have built and run the FHIRCore app to verify my change fixes the issue and/or does not break the app 
- [ ] I have checked that this PR does NOT introduce **breaking changes** that require an update to **_Content_** and/or **_Configs_**? _If it does add a sample here or a link to exactly what changes need to be made to the content._

**Code Reviewer Checklist**
- [ ] I have verified **Unit tests** have been written for any new feature(s) and edge cases
- [ ] I have verified any strings visible on UI components are in the `strings.xml` file
- [ ] I have verifed the [CHANGELOG.md](./CHANGELOG.md) file has any notable changes to the codebase
- [ ] I have verified the solution has been implemented in a configurable and generic way for reuseable components
- [ ] I have built and run the FHIRCore app to verify the change fixes the issue and/or does not break the app

### CODE REVIEW

File: android/gradle/libs.versions.toml [hunk 1]
```diff
@@ -20,12 +20,12 @@ cql-engine = "2.4.0"
...
 cql-to-elm = "2.4.0"
 dagger-hilt-compiler = "2.45"
-data-capture = "1.0.0-preview15-SNAPSHOT"
+data-capture = "1.0.0-preview18-SNAPSHOT"
 desugar-jdk-libs = "1.1.5"
 easy-rules-jexl = "4.1.0"
 espresso-core = "3.5.1"
 fhir-common-utils = "0.0.10-SNAPSHOT"
-fhir-engine = "0.1.0-beta03-preview10-SNAPSHOT"
+fhir-engine = "0.1.0-beta04-preview3.3-SNAPSHOT"
```
Comment (pld) [inline android/gradle/libs.versions.toml:28]: just to make sure I understand, are these Google or Ona releases?

Comment (ellykits) [reply android/gradle/libs.versions.toml:28]: They are Ona releases.

Comment (pld) [reply android/gradle/libs.versions.toml:28]: K, if Android FHIR SDK did a release from main today, could we upgrade to it, or do we have custom code, and/or code from a non-main in our releases?

Comment (ellykits) [reply android/gradle/libs.versions.toml:28]: Yes, it should be possible to upgrade to the SDK releases once all the unmerged PRs (in our release variant) are included in the official release. We keep track of the unmerged PRs, documented [in the remote-publishing section of the docs](https://docs.opensrp.io/engineering/android-app/developer-setup/publishing-fhir-sdk-artifacts#remote-publishing)

Comment (ellykits) [reply android/gradle/libs.versions.toml:28]: Also, NOTE that the `opensrp/android-fhir` master branch is always in sync with the `google/android-fhir` before we publish any of the SDK libraries.

Comment (pld) [reply android/gradle/libs.versions.toml]: k, is there a list of the unmerged PRs somewhere? I can't figure out where to find that

Comment (ellykits) [reply android/gradle/libs.versions.toml]: Here are the current unmerged changes PR included in the master-release branch https://github.com/opensrp/android-fhir/commit/master-release.

Comment (pld) [reply android/gradle/libs.versions.toml]: ah sweet, thank you!

**Review by pld — APPROVED** [review]

File: android/gradle/libs.versions.toml (hunk 1 above)
Comment (ellykits) [reply android/gradle/libs.versions.toml]: We can schedule an internal discussion together with @ndegwamartin to identify the gaps with the current setup to identify how best the SDK team can support us.

**Pull request merged**